File: segmentController/src/main/java/com/clumob/segment/controller/SegmentPagerItemController.java
package com.clumob.segment.controller;


import com.clumob.listitem.controller.source.ItemControllerImpl;
import com.clumob.listitem.controller.source.ItemUpdatePublisher;

/**
 * Created by prashant.rathore on 03/07/18.
 */
@Deprecated
public class SegmentPagerItemController extends ItemControllerImpl<SegmentInfo> {

    public SegmentPagerItemController(SegmentInfo viewModel) {
        super(viewModel);
    }

    @Override
    final public void onCreate(ItemUpdatePublisher publisher) {
        super.onCreate(publisher);
    }

    @Override
    final public void onAttach(Object source) {
        super.onAttach(source);
    }

    @Override
    final public void onDetach(Object source) {
        super.onDetach(source);
    }

    @Override
    final public void onDestroy() {
        super.onDestroy();
    }

    @Override
    public int getType() {
        return 0;
    }

    @Override
    final public long getId() {
        return viewData.getId();
    }

    public String getPageTitle() {
        return null;
    }
}

File: segmentController/src/main/java/com/clumob/segment/controller/SegmentPagerItemController.kt
package com.clumob.segment.controller

import com.clumob.listitem.controller.source.ItemUpdatePublisher
import com.clumob.segment.controller.list.SegmentItemControllerImpl

/**
 * Created by prashant.rathore on 03/07/18.
 */
@Deprecated("")
class SegmentPagerItemController(viewModel: SegmentInfo) : SegmentItemControllerImpl(viewModel) {

    override fun onCreate(publisher: ItemUpdatePublisher) {
        super.onCreate(publisher)
    }

    override fun onAttach(source: Any) {
        super.onAttach(source)
    }

    override fun onDetach(source: Any) {
        super.onDetach(source)
    }

    override fun onDestroy() {
        super.onDestroy()
    }

    override val type: Int
        get() = 0

    override val id: Long
        get() = segmentInfo.id.toLong()

    open val pageTitle: String?
        get() = null
}